CommonUI/Sources/CommonUI/MovieUI/MovieDetails.swift

```diff
@@ -21,8 +22,12 @@ public struct MovieDetails: View {
                     .frame(width: 30, height: 30)
                     .foregroundStyle(Color.red)
                     .onTapGesture {
-                        //TODO: Work on passing isFavorite here...
-                        print("⚠️ Update Favorite Data Here....Or should we do it outside of this view....")
+                        if isFavorite {
+                            UserData.shared.favoriteMovies.removeAll(where: { $0.imdbId == self.movie.imdbId })
```

Comment:
this needs to be in a MovieDetailsViewModel
the `.ontapGesture` should just call `.viewModel.heartWasTapped()` (or whatever)

and the entire if/else should be invoked inside that method
otherwise its not possible to test this

Reply:
when creating the `MovieDetailsViewModel` make sure to extract out the `UserData.shared` as a var at the top. This will allow testing to inject a Mock

Reply:
Where is the movieDetailsViewModel should be? Right now the MovieDetailsView is in CommonUI package. When I was building this MovieDetailsView, I though about this case. If we have MovieDetailsViewModel in the CommonUI package, it does not make so much sence to me. Or should we pass function to MovieDetailsView ?

Reply:
* create a dir `MovieDetail`
* rename `MovieDetail` file to `MovieDetailView`
* move file to `MovieDetail` dir
* create `MovieDeailViewModel`

Reply:
Ignore this comment

We will do this later in a "Quality of life" update